cpp/src/arrow/acero/CMakeLists.txt

```diff
@@ -129,6 +129,10 @@ if(ARROW_TESTING)
   if(ARROW_WITH_OPENTELEMETRY)
     target_link_libraries(arrow_acero_testing PRIVATE ${ARROW_OPENTELEMETRY_LIBS})
   endif()
+  # arrow_compute_testing will register the kernels for Gtest. In order to register the kernels
+  # for Google benchmark we use a custom main function used on add_arrow_compute_benchmark.
+  set(ARROW_ACERO_BENCHMARKS_TEST_LINK_LIBS
+      ${ARROW_ACERO_TEST_LINK_LIBS} arrow_acero_testing arrow_compute_core_testing)
```

Comment:
Shall we put this logic into `if(ARROW_BUILD_BENCHMARKS)` section down below so that the benchmarking stuff is more collocated?

Reply:
It seems `ARROW_TESTING` is a dependency for `ARROW_BUILD_BENCHMARKS`, we can use `ARROW_BUILD_BENCHMARKS` and `ARROW_BUILD_TESTS` independently but if any of those are used the arrow_testing libraries are built.
https://github.com/apache/arrow/blob/587b3fcad5ae4524411714c80a1479ff9a435508/cpp/cmake_modules/DefineOptions.cmake#L220-L240

Thanks for raising this concerns, I am learning a bunch of things I didn't know :)

Reply:
Ah sorry, I remember wrong. Thanks for clarifying!